core/src/main/java/com/datastax/oss/driver/internal/core/metadata/token/KeyspaceTokenMap.java

```diff
@@ -96,8 +96,11 @@ private KeyspaceTokenMap(
     return tokenRangesByNode.get(replica);
   }
 
-  Set<Node> getReplicas(ByteBuffer partitionKey) {
-    return getReplicas(tokenFactory.hash(partitionKey));
+  Set<Node> getReplicas(TokenFactory partitioner, ByteBuffer partitionKey) {
```

Comment:
Why not to take Partitioner instead of TokenFactory here and remove casting from DefaultTokenMap.getReplicas?